dht/dht.go

```diff
@@ -618,13 +648,28 @@ func (s *Server) nodeByID(id string) *node {
 	return nil
 }
 
-func (s *Server) handleQuery(source dHTAddr, m Msg) {
+func (s *Server) handleQuery(source DHTAddr, m Msg) {
+	var (
+		newmsg *Msg
+		skip   bool
+	)
 	node := s.getNode(source, m.SenderID())
 	node.lastGotQuery = time.Now()
 	// Don't respond.
 	if s.config.Passive {
 		return
 	}
+	// Call any hooks that may apply
```

Comment:
I thought about this; it's a question of option primacy, but also being consistent with what the options "promise". That is, "Passive" means "don't respond", but hooks could hypothetically respond, so that breaks what looks like a guarantee.

Keeping that guarantee would have required one of:

* Overriding hooks brutally (which is what I opted for, for simplicity)
* Changing response logic in a more dramatic way (so that if either routing logic or hooks called a response method, it would check and block there instead), which I felt would be a pretty big PR to issue first-time. :)

..or, discard the guarantee, and let hooks handle things before Passivity is checked. In this case, though I feel like "Passive" should be renamed to "DisableDefaultResponse" or something, which would make it clearer that it will still allow non-default responses through Hooks, while disabling what comes after.